### DISCUSSION TITLE
[Discussion] How to test failure cases?

### DISCUSSION THREAD

**Discussion by yourbuddyconner** [celo-org/optics-monorepo#665] [Ideas]
#659 is the epic relating to Testing of contract/agent failure cases in our `dev` testnet environment
- #660 is for testing Double Update
- #661 is for testing Improper Update

This thread is for discussion about exactly how we might go about doing this. 

Ideas: 
- Modify an updater agent to submit bad updates

**Comment by prestwich** [comment]
it should be pretty easy to write a typescript script that will deploy contracts and sign invalid bundles 

@ErinHales thoughts on this?

**Reply by ErinHales** [reply]
Ya, shouldn't be too hard.  I'm on it :)

**Reply by prestwich** [reply]
awesome. open some issues describing the work :)

**Reply by ErinHales** [reply]
Yeah, Conner already opened some issues so I'll work off of those
